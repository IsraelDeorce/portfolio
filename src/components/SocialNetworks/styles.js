import styled from 'styled-components'

export const SocialNetworks = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.c_brackets};
  & > * {
    cursor: pointer;
  }
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  & > *:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
