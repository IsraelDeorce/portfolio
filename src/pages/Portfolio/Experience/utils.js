import { SiHp, SiDell } from 'react-icons/si'
import { GiClapperboard, GiProgression } from 'react-icons/gi'
import pucrs from '../../../assets/pucrs.svg'
import ages from '../../../assets/ages.jpg'
export const projects = [
  {
    image: <SiHp color='#0096D6' />,
    title: 'HP Inc.',
    subtitle: 'subtitle',
    description: 'desc.hp'
  },
  {
    image: <SiDell color='#007DB8' />,
    title: 'Dell Inc.',
    subtitle: 'subtitle',
    description: 'desc.dell'
  },
  {
    image: <img src={pucrs} />,
    title: 'PUCRS',
    subtitle: 'desc.research_intern',
    description: 'desc.intern'
  },
  {
    image: <GiClapperboard color='#FF7F00' />,
    title: 'Gina Tells',
    subtitle: 'desc.personal_project',
    description: 'desc.ginatells'
  },
  {
    image: <GiProgression  color='#0E6B0E' />,
    title: 'Software KPIs',
    subtitle: 'desc.personal_project',
    description: 'desc.kpis'
  },
  {
    image: <img src={ages} />,
    title: 'AGES',
    subtitle: 'desc.nonprofit_engineer',
    description: 'desc.ages'
  }
]
