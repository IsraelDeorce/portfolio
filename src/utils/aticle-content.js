const articles = [
    {
        name: 'learn-react',
        title: 'The Fastest Way to Learn React',
        content: ['blablabla', 'blobloblo']
    },
    {
        name: 'learn-react2',
        title: 'The Fastest Way to Learn React2',
        content: ['blablabla2', 'blobloblo2']
    },
    {
        name: 'learn-react3',
        title: 'The Fastest Way to Learn React3',
        content: ['blablabla3', 'blobloblo3']
    }
]

export default articles