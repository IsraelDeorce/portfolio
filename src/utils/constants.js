export const LINKS = {
  GDRIVE: 'https://docs.google.com/document/d/1gwsxihd7lOVGqM2HzhD2VqOYW85ExDR7hLntG_dOQyg/edit?usp=sharing',
  WHATSAPP: 'https://wa.me/5551980254545',
  GITHUB: 'https://github.com/IsraelDeorce',
  LINKEDIN: 'https://www.linkedin.com/in/israeldeorce',
}

export const TEXT = {
  MYNAME: 'Israel Deorce',
  EMAIL: 'israel.idvj@gmail.com',
  PHONE: '+55 (51) 98025-4545',
  PORTO: 'Porto Alegre'
}
