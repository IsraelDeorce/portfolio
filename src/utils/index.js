export function formatLanguage (language) {
  if (typeof language === 'string') {
    if (language.length > 2) language = language.substring(0, 2)
    language = language.toLowerCase()
  }
  return language
}
