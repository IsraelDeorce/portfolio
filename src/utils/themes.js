const DarkTheme = {
  c_background: '#1E1E1E',
  c_menu: '#252526',
  c_tab: '#3C3C3C',
  c_string: '#CE9078',
  c_function: '#DCDCAA',
  c_varType: '#569CD6',
  c_varDeclaration: '#4FC1DB',
  c_varProperty: '#9CDCFE',
  c_brackets: '#D4D4C8',
  c_component: '#4EC9B0',
  c_number: '#A7CEA8',
  c_importExport: '#C586C0'
}

const LightTheme = {
  c_background: '#FFFFFF',
  c_menu: '#F3F3F3',
  c_tab: '#DDDDDD',
  c_string: '#A31515',
  c_function: '#795D26',
  c_varType: '#0000E1',
  c_varDeclaration: '#0071C1',
  c_varProperty: '#000F80',
  c_brackets: '#000000',
  c_component: '#267F5A',
  c_number: '#098658',
  c_importExport: '#AF00DB'
}

const Themes = {
  dark: DarkTheme,
  light: LightTheme
}

export default Themes
