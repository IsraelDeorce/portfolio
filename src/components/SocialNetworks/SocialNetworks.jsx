import { FaWhatsapp, FaGithub, FaLinkedin, FaEnvelope } from 'react-icons/fa'

import * as S from './styles'

function SocialNetworks () {
  return (
    <S.SocialNetworks>
      <FaWhatsapp size='32'/>
      <FaGithub size='32' />
      <FaLinkedin size='32' />
      <FaEnvelope size='32' />
    </S.SocialNetworks>
  )
}

export default SocialNetworks
