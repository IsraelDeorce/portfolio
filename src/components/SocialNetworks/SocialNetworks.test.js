import { LINKS } from '../../utils/constants'
import SocialNetworks from './SocialNetworks'

describe('<SocialNetworks/>', () => {
  beforeEach(() => jest.clearAllMocks())

  it('redirects to the whatsapp link when the whatsapp icon is clicked', () => {
    const whatsappIcon = shallow(<SocialNetworks/>).find('FaWhatsapp')
    global.open = jest.fn()
    whatsappIcon.simulate('click')
    expect(global.open).toHaveBeenCalledTimes(1)
    expect(global.open).toHaveBeenCalledWith(LINKS.WHATSAPP, '_blank')
  })

  it('redirects to the github link when the github icon is clicked', () => {
    const githubIcon = shallow(<SocialNetworks/>).find('FaGithub')
    global.open = jest.fn()
    githubIcon.simulate('click')
    expect(global.open).toHaveBeenCalledTimes(1)
    expect(global.open).toHaveBeenCalledWith(LINKS.GITHUB, '_blank')
  })

  it('redirects to the linkedin link when the whatsapp icon is clicked', () => {
    const linkedinIcon = shallow(<SocialNetworks/>).find('FaLinkedin')
    global.open = jest.fn()
    linkedinIcon.simulate('click')
    expect(global.open).toHaveBeenCalledTimes(1)
    expect(global.open).toHaveBeenCalledWith(LINKS.LINKEDIN, '_blank')
  })

  it('redirects to the email link when the whatsapp icon is clicked', () => {
    const envelopeIcon = shallow(<SocialNetworks/>).find('FaEnvelope')
    global.open = jest.fn()
    envelopeIcon.simulate('click')
    expect(global.open).toHaveBeenCalledTimes(1)
    expect(global.open).toHaveBeenCalledWith(LINKS.EMAIL, '_blank')
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<SocialNetworks />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<SocialNetworks />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<SocialNetworks />)).toMatchSnapshot()
    })
  })
})
