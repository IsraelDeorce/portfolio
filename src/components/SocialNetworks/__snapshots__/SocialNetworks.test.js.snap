// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialNetworks/> Snapshot renders without crashing using shallow 1`] = `
<styled.div>
  <a
    href="https://wa.me/5551980254545"
    target="_blank"
    title="https://wa.me/5551980254545"
  >
    <FaWhatsapp
      size="46"
    />
  </a>
  <a
    href="https://github.com/IsraelDeorce"
    target="_blank"
    title="https://github.com/IsraelDeorce"
  >
    <FaGithub
      size="46"
    />
  </a>
  <a
    href="https://www.linkedin.com/in/israeldeorce"
    target="_blank"
    title="https://www.linkedin.com/in/israeldeorce"
  >
    <FaLinkedin
      size="46"
    />
  </a>
  <a
    href="mailto: israel.idvj@gmail.com"
    target="_blank"
    title="israel.idvj@gmail.com"
  >
    <FaEnvelope
      size="46"
    />
  </a>
</styled.div>
`;
